src/unix/udp.c

```diff
@@ -954,8 +954,8 @@ static int uv__udp_set_source_membership6(uv_udp_t* handle,
     mreq.gsr_interface = 0;
   }
 
-  STATIC_ASSERT(sizeof(mreq.gsr_group) >= sizeof(*multicast_addr));
-  STATIC_ASSERT(sizeof(mreq.gsr_source) >= sizeof(*source_addr));
+  assert(sizeof(mreq.gsr_group) >= sizeof(*multicast_addr));
```

Comment:
I second @alzix's remark about the static assert. Apropos checking symbols at compile times: libuv generally tries to do feature detection at runtime, not build time. The build machine can be pretty different from the systems libuv actually runs on.

Reply:
> Apropos checking symbols at compile times: libuv generally tries to do feature detection at runtime, not build time. The build machine can be pretty different from the systems libuv actually runs on.

In such a cases cross-compiler is typically being used. Compiler must know target's machine ABI - otherwise code wont work properly. Static asserts are evaluated by the cross-compiler preprocessor and do not depend on the machine architecture used to run the build.
Asserts are invariants that should always be true. So if the condition is always true - it is also true at build time. Thus static asserts should be preferred over runtime asserts where applicable.

Reply:
I agree, but I'm doubly confused about your `CheckSymbolExists` comment now.

Reply:
Me too, lol.

Reply:
`STATIC_ASSERT` macro implements [`static_assert`](https://en.cppreference.com/w/cpp/language/static_assert) which may not be present in several toolchains.
The warnings actually comes from custom `static_assert` implementation. 
IMO `CheckSymbolExists` can help detecting if the symbol exists in compile time. If exists we can reuse it rather than provide custom implementation.

Reply:
I was wrong about `CheckSymbolExists` as it could not properly detect `static_assert` presence.
`check_c_source_compiles` does trick.
please see #3186